### ISSUE TITLE
PCF_IN_FCN_DEF not set when defining function pointers

### ISSUE THREAD

**Issue by micheleCTDE** [uncrustify/uncrustify#4048]
When defining a function pointer, it would be better to mark the parameters list as `PCF_IN_FCN_DEF`.
Note in the output below that this is not happening, while instead it happens when declaring a function.
This can help to indent/align code more consistently.
@guy-maurel what's your opinion?

Input file
```
void (*fnPtr)(int i, double d);
void function(int i, double d);
```

Config: tests/common/empty.cfg

First dump file
```
STEP: After uncrustify_start()
--------------
# -=====-
# number of loops               = 0
# -=====-
# language                      = C
# -=====-
# Line                Tag         Parent_type  Type of the parent         Columns Br/Lvl/pp             Flags   Nl  Text
#   1>               TYPE|           FUNC_VAR|     PARENT_NOT_SET[  1/  1/  5/  0][0/0/0][             c 0000][0-0] void
#   1>        TPAREN_OPEN|           FUNC_VAR|     PARENT_NOT_SET[  6/  6/  7/  1][0/0/0][        2 0300 0000][0-0]      (
#   1>           PTR_TYPE|               NONE|     PARENT_NOT_SET[  7/  7/  8/  0][0/1/0][        2 0008 0000][0-0]       *
#   1>           FUNC_VAR|               NONE|     PARENT_NOT_SET[  8/  8/ 13/  0][0/1/0][           308 0000][0-0]        fnPtr
#   1>       TPAREN_CLOSE|           FUNC_VAR|     PARENT_NOT_SET[ 13/ 13/ 14/  0][0/0/0][        2 0000 0000][0-0]             )
#   1>        FPAREN_OPEN|         FUNC_PROTO|     PARENT_NOT_SET[ 14/ 14/ 15/  0][0/0/0][        2 0000 0000][0-0]              (
#   1>               TYPE|               NONE|     PARENT_NOT_SET[ 15/ 15/ 18/  0][0/1/0][            88 0000][0-0]               int
#   1>               WORD|               NONE|     PARENT_NOT_SET[ 19/ 19/ 20/  1][0/1/0][           100 0000][0-0]                   i
#   1>              COMMA|               NONE|     PARENT_NOT_SET[ 20/ 20/ 21/  0][0/1/0][        2 0000 0000][0-0]                    ,
#   1>               TYPE|               NONE|     PARENT_NOT_SET[ 22/ 22/ 28/  1][0/1/0][            88 0000][0-0]                      double
#   1>               WORD|               NONE|     PARENT_NOT_SET[ 29/ 29/ 30/  1][0/1/0][           100 0000][0-0]                             d
#   1>       FPAREN_CLOSE|         FUNC_PROTO|     PARENT_NOT_SET[ 30/ 30/ 31/  0][0/0/0][        2 0000 0000][0-0]                              )
#   1>          SEMICOLON|           FUNC_VAR|     PARENT_NOT_SET[ 31/ 31/ 32/  0][0/0/0][        2 0000 0000][0-0]                               ;
#   1>            NEWLINE|               NONE|     PARENT_NOT_SET[ 32/ 32/  1/  0][0/0/0][                  0][1-0]
#   2>               TYPE|         FUNC_PROTO|     PARENT_NOT_SET[  1/  1/  5/  0][0/0/0][             c 0000][0-0] void
#   2>         FUNC_PROTO|               NONE|     PARENT_NOT_SET[  6/  6/ 14/  1][0/0/0][                  0][0-0]      function
#   2>        FPAREN_OPEN|         FUNC_PROTO|     PARENT_NOT_SET[ 14/ 14/ 15/  0][0/0/0][        2 0000 0000][0-0]              (
#   2>               TYPE|               NONE|     PARENT_NOT_SET[ 15/ 15/ 18/  0][0/1/0][            88 0008][0-0]               int
#   2>               WORD|               NONE|     PARENT_NOT_SET[ 19/ 19/ 20/  1][0/1/0][           100 0008][0-0]                   i
#   2>              COMMA|               NONE|     PARENT_NOT_SET[ 20/ 20/ 21/  0][0/1/0][        2 0000 0008][0-0]                    ,
#   2>               TYPE|               NONE|     PARENT_NOT_SET[ 22/ 22/ 28/  1][0/1/0][            88 0008][0-0]                      double
#   2>               WORD|               NONE|     PARENT_NOT_SET[ 29/ 29/ 30/  1][0/1/0][           100 0008][0-0]                             d
#   2>       FPAREN_CLOSE|         FUNC_PROTO|     PARENT_NOT_SET[ 30/ 30/ 31/  0][0/0/0][        2 0000 0008][0-0]                              )
#   2>          SEMICOLON|         FUNC_PROTO|     PARENT_NOT_SET[ 31/ 31/ 32/  0][0/0/0][        2 0000 0000][0-0]                               ;
#   2>            NEWLINE|               NONE|     PARENT_NOT_SET[ 32/ 32/  1/  0][0/0/0][                  0][2-0]
# -=====-
```

**Comment by gmaurel** [comment]
Yes!
It is always better to mark (anything) as soon as possible.

**Comment by micheleCTDE** [comment]
Great, will work on it and prepare a PR when done.

**Comment by micheleCTDEAdmin** [comment]
Fixed by PR #4051
